### PR TITLE
[+] Adds new endpoint for getting a users profile

### PR DESCRIPTION
**Pull request by Matt-Beldon-CDS**
[*] Updates getSiteUser to use the new user profile endpoint

contains a fix for one of the issues raised from #208

### CODE REVIEW

**Comment by andy-smith-cds** [comment]
@Matt-Beldon-CDS this is the one where we discussed the property names on the /update GET endpoint and the non-update GET endpoint so that they match up please? Then I can just conditionally add the /update on the end of the endpoint on this service depending on which page we're on. Cheers!

**Comment by andy-smith-cds** [comment]
@Matt-Beldon-CDS this is the one where we discussed the property names on the /update GET endpoint and the non-update GET endpoint so that they match up please? Then I can just conditionally add the /update on the end of the endpoint on this service depending on which page we're on. Cheers!

**Comment by Matt-Beldon-CDS** [comment]
> @Matt-Beldon-CDS this is the one where we discussed the property names on the /update GET endpoint and the non-update GET endpoint so that they match up please? Then I can just conditionally add the /update on the end of the endpoint on this service depending on which page we're on. Cheers!

Sorry got distracted by the registration stuff, will try get it sorted today

**Review by Tim-Hunt-CDS — APPROVED** [review]

**Comment by andy-smith-cds** [comment]
> > @Matt-Beldon-CDS this is the one where we discussed the property names on the /update GET endpoint and the non-update GET endpoint so that they match up please? Then I can just conditionally add the /update on the end of the endpoint on this service depending on which page we're on. Cheers!
> 
> Sorry got distracted by the registration stuff, will try get it sorted today

Sorry man we need the image to be returned on both endpoints too on the same property. Currently we only get the image returned on an 'image' property on the /update endpoint and on the non-update endpoint it returns 'profileImage' and it's null

**Review by JackWCDS — APPROVED** [review]

File: futurenhs.app/pages/users/[userId]/update/index.page.tsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ export const getServerSideProps: GetServerSideProps = withUser({
                      */
                     try {
                         const [userData] = await Promise.all([
-                            getSiteUser({ user, targetUserId }),
+                            getSiteUser({ user, targetUserId, isForEdit: true }),
```
Comment (ReallySmall) [inline futurenhs.app/pages/users/[userId]/update/index.page.tsx:91]: could we call this isForUpdate? I'm as guilty as anyone for this, but we have a combination of term uses between update and edit. I'd rather we try and stick to update so it matches the http verb it's usually used with in requests

**Review by ReallySmall — CHANGES_REQUESTED** [review]

File: futurenhs.app/pages/users/[userId]/update/index.page.tsx (hunk 1 above)
Comment (andy-smith-cds) [reply futurenhs.app/pages/users/[userId]/update/index.page.tsx:91]: @ReallySmall in the other places we're using this we have isForEdit - group members, groups, folders etc so I followed this pattern. Want me to change them all to isForUpdate?

Comment (andy-smith-cds) [reply futurenhs.app/pages/users/[userId]/update/index.page.tsx]: @ReallySmall Addressed this now, changed all instances to isForUpdate instead, matches the route & http verb as you say

**Review by ReallySmall — APPROVED** [review]

**Review by andy-smith-cds — APPROVED** [review]

**Pull request merged**